### PR TITLE
docs: add initial networking doc

### PR DESCRIPTION
**Pull request by corverroos**
Adds initial networking doc. To be expanded.

category: docs
ticket: #147

### CODE REVIEW

**Comment by OisinKyne** [comment]
I think we should be locating more of these docs here https://docs.obol.tech/docs/next/charon/peer-discovery or here https://docs.obol.tech/docs/next/charon/p2p-interface (or in a rethought doc that replaces/consolidates both)

File: docs/networking.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  - Charon is middleware DVT client and is therefore connected to an upstream beacon node and a downstream validator client is connected to it.
+  - Each operator should run the whole validator stack (all 4 client software types), either on the same machine or on different machines.
+  - The networking between the nodes should be private and not exposed to the public internet.
+  - Charon is configured to connect to the beacon node via the `--beacon-node-endpoints` flag.
+  - Charon is configured to listen and serve requests from the validator client via the `--validator-api-address` flag.
+- **External P2P Network**: `Charon <-> Charon <-> Charon <-> Charon`
+  - The Charon nodes in a DVT cluster are connected to each other via a small p2p network consisting of only the nodes in the cluster.
+  - Peer addresses are discovered via an external "relay" server. Obol hosts public relay server for this purpose, but anyone can host their own.
+  - The p2p networking is over the public internet so the charon p2p port must be publicly accessible.
+  - Charon is configured to listen and serve p2p requests via the `--p2p-tcp-addresses` flag.
```
Comment (xenowits) [inline docs/networking.md:18]: This reminds me - we should probably deprecate `--p2p-udp-address` flag in the next release

**Review by xenowits — APPROVED** [review]

File: docs/networking.md (hunk 1 above)
Comment (corverroos) [reply docs/networking.md:18]: already done

**Comment by corverroos** [comment]
See https://docs.google.com/document/d/1DKbXOlY8CU3saYAxjEdy8bcjYfGBaF87lfF4jKKFVtU/edit#

**Pull request closed without merging**